selftests/unit/test_task_info.py

```diff
@@ -0,0 +1,24 @@
+from unittest import TestCase
+
+from avocado.core.nrunner import Runnable, Task
+from avocado.core.task.info import TaskInfo
+
+
+class Info(TestCase):
+
+    def setUp(self):
+        runnable = Runnable('noop', 'noop')
+        task = Task('1-noop', runnable)
+        self.task_info = TaskInfo(task)
+
+    def test_empty(self):
+        self.assertIsNone(self.task_info.status)
```

Comment:
> IMO this usage is very strange. `task.task_info.status`, instead of a direct `task.status`. I understand that at some point tasks will not have all information, but as I said before in a previous review, from the OOP perspective there is no problem with this being part of `Task`. I can't see the reason for this separation.

Do you think a task should know that it's supposed to be started by a spawner? And that it should have a place for the spawner's own handling? I find this very wrong.  You mention the "OOP perspective", but composition is widely used in OOP. 

> 
> We are creating a new class because only 4 attributes that are related to a Task.

Right, and I think they're in the right place.  If you look at how a task *can be run*, say, `Task.run()` or `avocado-runner-* task-run ...` it needs none of the attributes.  So they're realted to a Task, but they're not part of a Task.

Reply:
> > IMO this usage is very strange. `task.task_info.status`, instead of a direct `task.status`. I understand that at some point tasks will not have all information, but as I said before in a previous review, from the OOP perspective there is no problem with this being part of `Task`. I can't see the reason for this separation.
> 
> Do you think a task should know that it's supposed to be started by a spawner? And that it should have a place for the spawner's own handling? I find this very wrong. You mention the "OOP perspective", but composition is widely used in OOP.
> 

I can't see why not. And you find this very wrong. Yes, the composition is widely used but I don't see the need here, at least how it is described in this PR. I'm also not saying that the current model is wrong from the OOP perspective.

When you do things like:

```python
is_task_alive(task_info)
spawn_task(task_info)
```
IMO, would be better if we had:

```python
is_task_alive(task)
spawn_task(task)
```

> > We are creating a new class because only 4 attributes that are related to a Task.
> 
> Right, and I think they're in the right place. If you look at how a task _can be run_, say, `Task.run()` or `avocado-runner-* task-run ...` it needs none of the attributes. So they're realted to a Task, but they're not part of a Task.

When you describe all 4 items (`status`, `execution_timeout`, `spawner_handle` and `spawning_result`) you mention the word "task" in the docstrings. So yes, I can see this as `task.status`, `task.execution_timeout`. Without a problem.

Reply:
Well, there's certainly an impasse here, and that's fine because "wisdom lives in the multitude of opinion".  While `is_task_alive(task)` is certainly an improvement over `is_task_alive(task_info)` for readability purposes, the ownership of the extra data is misplaced.

If we combine `TaskInfo` attributes within a `Task`, the following would happen:
 
```python
task.status = 'CHECKED REQUIREMENTS OK on podman image "foobar"'
PodmanSpawner.spawn_task(task)
cmd = podman_bin + "run" + image + task.get_args() # serialization into args
```

Within the container, the task:

```python
task = Task.from_commandline_args() # deserialization
# Task is alive and running
task.status == None # Nothing here?
task.execution_timeout == None # can I set it? no, not a test timeout, not handled by the Task itself... 
task.spawner_handle == None # No one spawned me?
```

And, if one is to write a `Task` implementation (as part of a `avocado-runner-*`) in any other language (which is absolutely a goal here), these extra attributes at the reference (Python) implementation would only to the development overhead or confusion, as it would make no sense it all within that context.

Reply:
Hi @clebergnu First of all: I don't want to be pokey here :) This just sounds strange to me and I'm trying to make this consistent. We are losing a lot with readability and maintainability, IMO. but feel free to ignore and move on, we can discuss this after LTS. Said that...
 
I'm not seeing the big picture here. But one problem that I was expecting to be fixed was this task des/ser from the command line. I thought that we had discussed this before, and we agreed that this was not so good. Probably I didn't understand properly.

IIUC, you are saying that we need a "composition" because we don't need a few attributes when deserializing? This doesn't feel the right argument for needing a composition. What if we use inheritance here?

BasicTask
CompleteTask(BasicTask):
   + plus those extra args?

In that way, we can have `task.status` (where `task` is a `CompleteTask()`) (The names are just suggestions)

Reply:
> Hi @clebergnu First of all: I don't want to be pokey here :) This just sounds strange to me and I'm trying to make this consistent. We are losing a lot with readability and maintainability, IMO. but feel free to ignore and move on, we can discuss this after LTS. Said that...
> 
> I'm not seeing the big picture here. But one problem that I was expecting to be fixed was this task des/ser from the command line. I thought that we had discussed this before, and we agreed that this was not so good. Probably I didn't understand properly.
> 

What do you mean by fixed? I don't really remember what the core issue is.

> IIUC, you are saying that we need a "composition" because we don't need a few attributes when deserializing? This doesn't feel the right argument for needing a composition. What if we use inheritance here?
> 
> class BasicTask:
> class CompleteTask(BasicTask):
> 
> * plus those extra args?
> 
> In that way, we can have `task.status` (where `task` is a `CompleteTask()`) (The names are just suggestions)

It doesn't look like we'll agree here.  I find this approach much worse for reasons explained [here](https://github.com/avocado-framework/avocado/pull/4128#discussion_r475724946) and also for the fact that `Tasks` are created much earlier, during the suite creation.  The old `avocado nlist` command, for instance, had a `--write-recipes-to-directory` which would persist those Tasks.  And the current `avocado list --resolver` won't ever attempt to deal with their execution (and could/should soon have a similar `--write-recipes-to-directory`). 

So we either favor creating the `CompleteTask` in most places or we do something like `CompleteTask.from_basic_task(task)`.

So, to summarize, I'm not convinced inheritance is the best solution here.

Reply:
I see, yes, the inheritance was just a suggestion to this current approach. No worries.

For me the strange part is:

"Where should I get 'Task information'?" In my mind, the answer is straightforward: All information related to a task should be inside the Task. Having another "TaskInfo" sounds very strange. And it is even strange when I do thinks like `is_task_alive(task_info)` or `spawn_task(task_info)`, as I said before.

But ok, probably we are not going to agree here. No worries, life continues. ;)

Reply:
I think this will be easier to resolve in the future where the actual use cases establish what is more appropriate way of structuring the data. So as it might be unclear for all sides, let's just give it some time and see where the implementation and its usage bring us.

Reply:
So the latest idea/proposal is to rename `TaskInfo` to `RuntimeTask`, so that the current `Task` can be continued to be used as is, and something like this:

```python
task_to_be_executed = RuntimeTask(task)
```

How does it sound?

Reply:
I think that is better. :) Thanks a lot.

Reply:
If I recall correctly, we discussed that the status of a task is not part of the task, but part of the state where the task is at one specific moment. I think the confusion here is more related to the structure naming than anything else.

Reply:
Yes, that's true.  Seems to be solved by the rename! :)